fault/spice_target.py

```diff
@@ -10,15 +10,32 @@
 from fault.psf_parse import psf_parse
 from fault.subprocess_run import subprocess_run
 from fault.pwl import pwc_to_pwl
-from fault.actions import Poke, Expect, Delay
+from fault.actions import Poke, Expect, Delay, Print
+from fault.select_path import SelectPath
+
+
+# define a custom error for A2D conversion to make it easier
+# to catch this specific issue (for example, in adapting
+# tests based on the previous test result)
+class A2DError(Exception):
+    pass
+
+
+class CompiledSpiceActions:
+    def __init__(self, pwls, checks, prints, stop_time, saves):
+        self.pwls = pwls
+        self.checks = checks
+        self.prints = prints
+        self.stop_time = stop_time
+        self.saves = saves
```

Comment:
Python 3.7's dataclass is a new feature that makes writing classes like this simpler (assuming your intended use pattern fits into the model), check out
* https://realpython.com/python-data-classes/
* https://docs.python.org/3/library/dataclasses.html